parser_test.go

```diff
@@ -256,12 +256,28 @@ func (s *testParserSuite) RunTest(c *C, table []testCase) {
 		parser.EnableWindowFunc()
 	}
 	for _, t := range table {
-		_, err := parser.Parse(t.src, "", "")
+		stmts, err := parser.Parse(t.src, "", "")
 		comment := Commentf("source %v", t.src)
-		if t.ok {
-			c.Assert(err, IsNil, comment)
-		} else {
+		if !t.ok {
 			c.Assert(err, NotNil, comment)
+			continue
+		}
+		c.Assert(err, IsNil, comment)
+		// restore correctness test
+		for _, stmt := range stmts {
+			var sb strings.Builder
+			err := stmt.Restore(&sb)
+			if err != nil {
```

Comment:
1. When we panic in Restore, we can use type assertion here to check whether the Node has implemented the Restore function.
2. We also need a test function like `TestRestore` to test whether the AstToSqlText result is as expected.

Reply:
1. I'm afraid that this approach will lead to problems in community cooperation.
2. We put it in `parser_test.go/RunTest`, and we add a unit test for one node.
like: https://github.com/pingcap/parser/pull/56/files#diff-77f9c5478bd9b2f3d33089e2ef1f4d90R106